website/static/js/formViewModel.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Maintains the base class for knockoutJS form ViewModels
+ */
+'use strict';
+
+var ko = require('knockout');
+
+var $osf = require('osfHelpers');
+var oop = require('js/oop');
+
+var ValidationError = oop.extend(Error, {
+    constructor: function (messages, header, level) {
+        this.super.constructor.call(this);
+        this.messages = messages || [];
+        this.header = header || 'Error';
+        this.level = level || 'warning';
+    }
+});
+
+var FormViewModel = oop.defclass({
```

Comment:
Please add a brief docstring. Most importantly, users of this class should know that they must implement `isValid`.